src/PrestaShopBundle/ApiPlatform/OpenApi/Adapter/SchemaSynchronizer.php

```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\ApiPlatform\OpenApi\Adapter;
+
+use ApiPlatform\Metadata\Operation;
+use ArrayObject;
+use PrestaShopBundle\ApiPlatform\DomainObjectDetector;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionMethod;
+use ReflectionProperty;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractorInterface;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface;
+use Throwable;
+
+/**
+ * Synchronizes the OpenAPI schema with the actual properties available in the resource.
+ * This removes fields that are documented but don't exist in the actual resource,
+ * and ensures properties from the resource are present in the schema.
+ */
+class SchemaSynchronizer implements OpenApiSchemaAdapterInterface
+{
+    public function __construct(
+        protected readonly PropertyInfoExtractorInterface $propertyInfoExtractor,
+        protected readonly ClassMetadataFactoryInterface $classMetadataFactory,
+        protected readonly DomainObjectDetector $domainObjectDetector
+    ) {
+    }
+
+    public function adapt(string $class, ArrayObject $definition, ?Operation $operation = null): void
+    {
+        if (!class_exists($class)) {
+            return;
+        }
+
+        if (empty($definition['properties'])) {
+            $definition['properties'] = new ArrayObject();
+        }
+
+        if (!$operation) {
```

Comment:
I understand the needed for the second part of this adapter: "if a property is set on the command but not on the API resource we remove it from the doc"

But in this case there is no operation, so no CQRS command, so I guess the class used is the API resource itself no? Then it's only for read operations

So my question is, do we need to check if the properties match the API resource properties in this case? Because my understanding is that that's already what is done by API platform, so the properties should already be synced no?